add some info on default sub config (#6801)
ArduPilot/ardupilot_wiki: sub/source/docs/sub-hardware.rst

.. _sub-hardware:

==================================
Default Sub Hardware Configuration
==================================

While the Sub firmware is currently being generalized to allow more flexibility, such as RC control, re-arrangement of outputs and inputs, etc.,
current code is targeted for a specific ROV configuration with the following hardware setup:

- Control is via a surface tether from a ground control station via MAVLink commands to the Sub's autopilot telemetry port, usually employing a joystick and multiple buttons for mode changes and peripheral hardware control. Control functions and autopilot outputs for thrusters and peripheral hardware have specific defaults (see :ref:`Sub defaults <sub-defaults>` below)
- A depth sensor (underwater barometer) is required.
- Frame configuration is defaulted to Vectored with 6 thrusters.

.. _sub-defaults:

Sub Defaults
============

* Default :ref:`frame type <FRAME_TYPE>` is VECTORED

> * RC is disabled via :ref:`RC_PROTOCOLS<RC_PROTOCOLS>` = 0 and :ref:`FLTMODE_CH<FLTMODE_CH>` = 0.

Reviewer: ```suggestion
* Radio control is disabled via :ref:`RC_PROTOCOLS<RC_PROTOCOLS>` = 0 and :ref:`FLTMODE_CH<FLTMODE_CH>` = 0. Water absorbs radio waves, so RC is generally impractical unless the vehicle is at the surface, or the signal is converted to some other medium in between (e.g. using a tether cable, or acoustic modems).
```

- People starting out with a Sub vehicle don't necessarily know what RC is.
- People gaining first exposure to underwater vehicles may not be aware of its electromagnetic absorption spectrum
    - Longer term I'm considering including a broader "challenges of the underwater environment" section to explain things like this, which this could link to, but it seems alright to include a relevant note here for now.

Author: I like it

MAVLink Overrides
-----------------

> Since pilot directional control, normally provided by RC in other vehicles is not enabled by default. Sub uses RC overrides via MAVLink, and the channel mapping is fixed:

Reviewer: ```suggestion
ArduPilot vehicles normally use radio control (RC) to provide pilot directional control. Since Sub has RC disabled by default, it instead provides directional control with MAVLink RC overrides, and the channel mapping is fixed:
```

Author: thanks

===============  ===================
Pilot control    RC channel override
===============  ===================
Roll             1
Pitch            0

> Yaw              3
> Throttle(Depth)  2
> Forward          4
> Lateral          5

Reviewer: ```suggestion
Yaw (Turn)       3
Heave (Vertical) 2
Surge (Forward)  4
Sway (Lateral)   5
```
Adjustments to better match [the firmware naming](https://github.com/ArduPilot/ardupilot/blob/master/libraries/SRV_Channel/SRV_Channel.cpp#L114), which links to [ship movement terminology](https://en.wikipedia.org/wiki/Ship_motions).

Author: maybe put the nautical after the conventional...will change

Camera Pan       6 (via AUX function on RC7)

> Camera Tilt      7 (via AUX function on RC8)

Reviewer: Not certain how best to incorporate this, but the old [ArduSub RC docs](https://www.ardusub.com/developers/rc-input-and-output.html) clarify that

> *Note: Camera Tilt control is not standard servo-style control. Specifying a 'PWM' value sets the speed for a sweep from the current tilt angle to the min/max camera angle. The input difference from the midpoint (1500) sets the speed, e.g. a 1900 input sweeps to the max camera angle twice as fast as 1700.

which seems sufficiently unobvious that it's worth including a note about.

Reviewer: Huh, I learnt something new - thanks! :-)

Lights1          8
Lights2          9
Video Switch     10
===============  ===================

Output Assignments
------------------
The default output assignments are:

======       ========
Output       Function
======       ========
1            Motor1
2            Motor2
3            Motor3
4            Motor4
5            Motor5
6            Motor6

> 9            RCin9

Reviewer: This recently got renamed to `RCIN9/Lights1Level` in the firmware, but I've just checked and there are seemingly dedicated outputs for lights now instead.

@Williangalvani can you confirm whether this should be changed to `Lights1`, or remain as `RCIN9`?

Author: this matches what the GCS reports

10           Mount1 Pitch
======       ========

Button Assignments
------------------

Default Sub uses GCS joystick buttons to implement mode switching, light activation, etc.

======      ========
Button      Function
======      ========
1           Mode Manual
2           Mode Depth_Hold
3           Mode Stabilize
4           Disarm
5           Shift Function (when pressed buttons provide alternate second function)
6           Arm
7           Center Mount1
8           Joystick Input Hold Set

> 9           Tilt Mount1 Down/Pan Mount Left
> 10          Tilt Mount1 Up/Pan Mount Right
> 11          Joystick Sensitivity Gain Increase/ Trim Pitch Decrease
> 12          Joystick Sensitivity Gain Decrease/ Trim Pitch Increase
> 13          Lights1 Dimmer/ Trim Roll Decrease
> 14          Lights1 Brighter/ Trim Roll Increase

Reviewer: ```suggestion
9           Tilt Mount1 Down / Pan Mount1 Left
10          Tilt Mount1 Up / Pan Mount1 Right
11          Joystick Sensitivity Gain Increase / Trim Pitch Decrease
12          Joystick Sensitivity Gain Decrease / Trim Pitch Increase
13          Lights1 Dimmer / Trim Roll Decrease
14          Lights1 Brighter / Trim Roll Increase
```

Author: actually we are both wrong...there should be no white space around the slash when it indicates an alternative....will correct

Reviewer: Is that an ArduPilot convention?

I read spacing around slashes as "word/alternative" / "option alternative".

Author: its a general grammar/punctuation rule...Goggle "whitespace around slashes":

"Generally, whitespace is not used around slashes (/) when they separate single words or letters. However, a space is typically used before and after the slash when it separates phrases, sentences, or a single word from a phrase, especially to improve readability. 
"

Reviewer: That's exactly what I was trying to suggest 😅 

As an example, it should be "Tilt Mount1 Down" / "Pan Mount1 Left" (i.e. two phrases being separated). If it's separating just the words in the middle (e.g. "Tilt Mount1 Down/Pan Mount1 Left") then it comes across as those words being the differentiator of two broader options (e.g. "Tilt Mount1 Down Mount1 Left" / "Tilt Mount1 Pan Mount1 Left").

As a stripped down example, "Lights 1/2" makes sense, and "expands" to "Lights 1 / Lights 2", but "Lights 1/Lights 2" is hard to make sense of.

Author: yep, I see your point...will fix
